### PR TITLE
Refactor `authorize.js` props and Add `alt` text to OAuth Related Images

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to the comments https://github.com/opencollective/opencollective-frontend/pull/7966#discussion_r912813689 and https://github.com/opencollective/opencollective-frontend/pull/7966#discussion_r912815870

### CODE REVIEW

File: components/SignIn.js [hunk 1]
```diff
@@ -109,12 +109,12 @@ export default class SignIn extends React.Component {
...
                 </Box>
                 <Box ml={24} mr={24} mt={32} minWidth={40}>
-                  <Image src="/static/images/oauth-flow-connect.png" alt="OAuth Connect" height={40} width={40} />
+                  <Image src="/static/images/oauth-flow-connect.png" height={40} width={40} />
                 </Box>
                 <Box minWidth={104}>
                   <img
                     src={this.props.oAuthAppImage}
-                    alt="OAuth Logo"
+                    alt="Logo of open authorize application"
```
Comment (znarf) [inline components/SignIn.js:117]: I don't see the value of this alt in term of accessibility. Better remove it.

Comment (znarf) [reply components/SignIn.js]: What about the one just before "Open Collective logo", how is it useful for accessibility?

Comment (SudharakaP) [reply components/SignIn.js]: I've removed that as well. I realized that we have lots of places that the `Open Collective Logo` string is used as alt strings. I've opened a new PR to remove all of them; https://github.com/opencollective/opencollective-frontend/pull/8005

**Review by znarf — CHANGES_REQUESTED** [review]
Conflict to fixes

**Comment by SudharakaP** [comment]
> Conflict to fixes

I've re-based it. For future I like to enable conflict notifications on slack for myself so I know when a conflict occurs. I've opened a discussion on this here; https://opencollective.slack.com/archives/C0RMV6F8C/p1657632806162659.

**Pull request merged**